### DISCUSSION TITLE
Is there a way to refresh snapshots?

### DISCUSSION THREAD

**Discussion by Artur-** [redhat-developer/vscode-java#2137] [Q&A]
When developing something involving multiple projects with a setup like

Project A provides some feature
Project B uses a snapshot of project A

You quickly run into the problem that you have one VS Code window open for each project, you fix something in project A and do `mvn install`.

You are able to run project B and it will use the new snapshot correctly but if you have a class from project A open e.g. for debugging, you will keep seeing the old source code so code lines won't match etc. Effectively you cannot debug.

The only workaround for this I have found is to do `Reload window` as it clears some cache(?) and then shows the correct sources for project A classes.

Is there a better way?

**Comment by fbricon** [comment]
Have you tried right clicking in B/pom.xml -> "Update project"?

<img width="989" alt="Screenshot 2021-09-29 at 10 49 55" src="https://user-images.githubusercontent.com/148698/135235555-e9183e34-1667-4d00-ac59-32d0d3dc8d14.png">

**Comment by Artur-** [comment]
Thanks. that also works! It is a bit easier than reloading the window but it still requires re-opening the class file afterwards. 

I thought that action was the same as
![image](https://user-images.githubusercontent.com/260340/135239759-34df5ed0-6644-437e-8ba8-daa4dc00300e.png)
but only "Update project" in the pom.xml works in this case.

What would be awesome is that I could either do "refresh" on the class file by right clicking or if the "Update project" command worked so that it would automatically refresh relevant open class files

**Comment by fbricon** [comment]
yes it's the same command in the pom.xml context menu and the command palette.
There's no way to trigger a classpath update from an opened class, because that class doesn't know it's part of a snapshot or that it's maintained by a build system allowing snapshot updates.

**Reply by fbricon** [reply]
However, we **might** be able to reload the opened .class files, after a classpath update event. Could you open an enhancement request?
